src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -859,11 +868,11 @@ public Task CloseAsync()
 #else
         public override Task CloseAsync()
 #endif
-            => Close(connectionClosing: false, async: true);
+            => Close(connectionClosing: false, async: true, isDisposing: false);
 
-        internal async Task Close(bool connectionClosing, bool async)
+        internal async Task Close(bool connectionClosing, bool async, bool isDisposing)
         {
-            if (State == ReaderState.Closed)
+            if (State == ReaderState.Closed || State == ReaderState.Disposed)
```

Comment:
This means that if we close a reader and then dispose it, it stays in the closed state (and e.g. throws the incorrect exception). That isn't a big bug, but we can easily fix it.

Reply:
But we do set it in the dispose, no?

Reply:
Ah yes, forgot about the finally.